### PR TITLE
Made collision only dodge closes part, better food handling.

### PR DESCRIPTION
**Pull request by chancity**
## Description

The bot will now go to larger food clusters, if large enough it will accelerate. Collision radius increases when accelerating
## Motivation and Context

The snake is hungry and now it can eat more
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by ermiyaeskandary** [comment]
Great - PR template works . I can test in 10 min. Can someone test ?

**Comment by ChadSki** [comment]
Working great for me!

File: bot.user.js [hunk 1]
```diff
@@ -311,17 +311,19 @@ document.onkeydown = function (e) {
             window.savePreference('collisionDetection', window.collisionDetection);
         }
 
-        // Letter 'A' to increase collision detection radius
-        if (e.keyCode === 65) {
+          // Letter 'A' to increase collision detection radius
+        if (e.keyCode === 65 && window.snake.sp < 10) {
```
Comment (ChadSki) [inline bot.user.js]: Is there a technical reason to not allow adjusting collision radius while accelerated?

**Comment by Drflash55** [comment]
This bot lags for me whenever it's trying to speed towards food...

It just totally freezes when it detects a huge cluster of food and speeds towards it, and the game resumes when my snake dies...

**Comment by chancity** [comment]
@Drflash55 it is probably the foodNearFood function causing that

anyone have any suggestions on how we could rewrite this method so its less CPU intensive?

**Comment by Seple** [comment]
Please optimize all of the code for the bot, so it's less-intensive on the CPU and gives decent FPS.

_Fixed up some of the grammar issues for you. ~ Drflash55_

**Comment by ChadSki** [comment]
> anyone have any suggestions on how we could rewrite this method so its less CPU intensive?

I will take a look.

**Comment by ChadSki** [comment]
Here's one place we can optimize:

```
// Sort the food based on their distance relative to player's snake
window.sortedFood = window.getSortedFood();
// Current food    
window.currentFood = window.sortedFood[0]; 
```

We only need to get the closest food. It's wasted effort to completely sort the list.

**Pull request closed without merging**